### DISCUSSION TITLE
ExchangeSharp vs CCXT NET

### DISCUSSION THREAD

**Discussion by GustavoEHennemann** [DigitalRuby/ExchangeSharp#716] [General]
Hello,
I'm starting a new project to connect to different crypto exchanges, and looking in internet I found ExchangeSharp. I checked for some features in the project git but I can't get deeper unless I start to use it or asking some else. So, does anyone knows CCXT NET and can share some experiences or compare it to ExchangeSharp? Which one can connect more exchanges, faster, harder to use, etc.? Anything helps.
My project uses winforms with .net5, maybe .net6 in the future.
Thanks in advance

**Comment by StarringLara** [comment]
What is nice about .NET is it is easy to try things out by adding it with a package manager. VS lets you do this through the GUI. I was able to start trying this library out and immediately start interacting with the public parts of a couple exchange APIs by referencing the code in the console app as sample code.

The hardest part about using ExchangeSharp so far for me is figuring out how to properly encrypt your API keys so you can load them, which you will need to do to work with the private part of APIs (initiating trade orders,  etc). There aren't many helpful instructions about how to setup your keys, you basically have to dig through the code itself to figure out what it is needing, which I am still currently figuring out. According to the README, you are supposed to use the console app to encrypt your keys, but it doesn't explain how to do this, nor does the console app help explain it. [Here](https://github.com/jjxtra/ExchangeSharp/issues/619) is another person stuck on this.

As far as CCXT.NET goes, I have never tried it, but I'm weary of it because it doesn't represent itself very well. The README misspells "Relase". When it explains how to install it, there is no mention of using a package manager, and they tell you to clone it into your project directory. Cloning one project into another is not how you are supposed to install a library in C#, so it seems they don't really know what they are doing. It also doesn't properly explain what it is. It says it is an extension of ccxt, but it doesn't explain what they mean by that. Ccxt does not support C#, thus the need for CCXT.NET. But I don't know if CCXT.NET started out as a C# port of ccxt, or if it uses ccxt under the hood. If it uses it under the hood, are they keeping up with the latest version of CCXT or have they fallen behind? How do we install them both together such that one can use the other? If it does use it under the hood, that seems like an extra layer of complexity that will lead to brittleness I don't want to deal with as a library consumer. It doesn't support Coinbase or KuCoin, which makes it seem not very up to date. Also the last release was on Aug 28 of 2020, so it probably doesn't even work with many of the exchange APIs by now, considering how fast everything is moving in crypto.

**Reply by vslee** [reply]
Try using `IExchangeAPI.LoadAPIKeysUnsecure()`

**Comment by jjxtra** [comment]
CryptoUtility.SaveUnprotectedStringsToFile. Then you can load them securely later. But if you don't care then vslee's method will work great.
